packages/itwinui-react/src/core/Carousel/Carousel.tsx

```diff
@@ -61,52 +61,11 @@ const CarouselComponent = React.forwardRef((props, ref) => {
     scrollToSlide.current(userActiveIndex, {
       instant: justMounted.current,
     });
-
-    // re-focus the carousel for keyboard nav, but not on first mount
-    // because it shows outline and might interfere with other components
-    if (!justMounted.current) {
-      carouselRef.current?.focus({ preventScroll: true });
-    }
-
     justMounted.current = false;
   }, [userActiveIndex]);
 
   const [slideCount, setSlideCount] = React.useState(0);
 
-  const [keysPressed, setKeysPressed] = React.useState<Record<string, boolean>>(
-    {},
-  );
-
-  const handleKeyDown = (event: React.KeyboardEvent) => {
-    if (event.altKey || event.ctrlKey || event.metaKey || event.shiftKey) {
-      return;
-    }
-
-    const key = event.key;
-    if (key === 'ArrowLeft' || key === 'ArrowRight') {
-      setKeysPressed((old) => ({ ...old, [key]: true }));
-    }
-  };
-
-  const handleKeyUp = (event: React.KeyboardEvent) => {
```

Comment:
Removing this broke switching slides when focus is on the slide.

Reply:
shouldn't be a problem. APG example behaves the same way: https://www.w3.org/WAI/ARIA/apg/patterns/carousel/examples/carousel-2-tablist/

Reply:
On the html demo page, when focusing the slide instead of the dots, and then pressing the arrow keys, the slide moves, but the dots do not update.

[15.webm](https://github.com/iTwin/iTwinUI/assets/45748283/0ae09af3-01c6-49a6-9ad0-fd1337e83193)

To reproduce:
* Press <kbd>Tab</kbd> to switch focus from the dots to the slide
* Press <kbd>←</kbd>/<kbd>→</kbd>

To confirm, is this only on my end?

Reply:
the HTML page is not meant to show the behavior, only the visuals. you should test the React components in storybook/playgrounds/website.

Reply:
It's really broken now. I would think to disable slide switching when focused on slide if we don't want to fix it.


https://github.com/iTwin/iTwinUI/assets/83585998/097f4952-64b3-4f16-b274-8005bcc7922a

Reply:
What's happening: focus on slide > click right arrow > it breaks > click tab > focuses slide again

Reply:
arrow keys are used for reading characters in NVDA so it makes sense that it's spelling out "A c t i o n s" once you exit the carousel in storybook

![](https://github.com/iTwin/iTwinUI/assets/9084735/80eb1921-a5ae-44ab-9963-e70ca638e2bc)

https://dequeuniversity.com/screenreaders/nvda-keyboard-shortcuts